src/BigDecimal.ts

```diff
@@ -0,0 +1,717 @@
+/**
+ * This module provides utility functions and type class instances for working with the `BigDecimal` type in TypeScript.
+ * It includes functions for basic arithmetic operations, as well as type class instances for `Equivalence` and `Order`.
+ *
+ * @since 2.0.0
+ */
+
+import * as BigI from "./BigInt"
+import * as Equal from "./Equal"
+import * as equivalence from "./Equivalence"
+import { dual } from "./Function"
+import * as Hash from "./Hash"
+import { type Inspectable, NodeInspectSymbol, toString } from "./Inspectable"
+import * as Option from "./Option"
+import * as order from "./Order"
+import type { Ordering } from "./Ordering"
+import { type Pipeable, pipeArguments } from "./Pipeable"
+
+const TypeId: unique symbol = Symbol.for("effect/BigDecimal")
+
+/**
+ * @since 2.0.0
+ * @category symbol
+ */
+export type TypeId = typeof TypeId
+
+/**
+ * @since 2.0.0
+ * @category models
+ */
+export interface BigDecimal extends Equal.Equal, Pipeable, Inspectable {
+  readonly [TypeId]: TypeId
+  readonly value: bigint
+  readonly scale: number
+}
+
+const BigDecimalProto: Omit<BigDecimal, "value" | "scale"> = {
+  [TypeId]: TypeId,
+  [Hash.symbol](this: BigDecimal): number {
+    return Hash.structure({ value: this.value, scale: this.scale })
```

Comment:
You will need to first normalize the value, otherwise you will get false negatives using Equal.equals.